_examples/json/main.go

```diff
@@ -0,0 +1,109 @@
+package main
+
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"os"
+	"reflect"
+
+	"github.com/alecthomas/kong"
+	"github.com/alecthomas/participle"
+)
+
+type pathExpr struct {
+	Parts []part `@@ { "." @@ }`
+}
+
+type part struct {
+	Obj string `@Ident`
+	Acc []acc  `("[" @@ "]")*`
+}
+
+type acc struct {
+	Name  *string `@(String|Char|RawString)`
+	Index *int    `| @Int`
+}
+
+var (
+	parser = participle.MustBuild(&pathExpr{})
+	cli    struct {
+		File string `arg:"" type:"existingfile" help:"File to parse."`
+	}
+)
+
+func main() {
+	ctx := kong.Parse(&cli)
+	r, err := os.Open(cli.File)
+	ctx.FatalIfErrorf(err)
+	defer r.Close()
+
+	var input map[string]interface{}
+	err = json.NewDecoder(r).Decode(&input)
+	ctx.FatalIfErrorf(err)
+
+	enc := json.NewEncoder(os.Stdout)
+	enc.SetIndent("", "  ")
+
+	for {
+		// Example: check_run.check_suite.pull_requests[0].head.sha
+		fmt.Fprint(os.Stderr, "Enter a json path: ")
+		q, err := bufio.NewReader(os.Stdin).ReadString('\n')
+		ctx.FatalIfErrorf(err)
+
+		var expr pathExpr
+		if err := parser.ParseString(q, &expr); err != nil {
+			fmt.Fprintln(os.Stderr, err)
```

Comment:
Use `ctx.FatalIfErrorf(err)` here.

Reply:
The idea previously was that the app should not exit on error, but instead ask for a new query. Now however as it's no longer interactive, i removed the kong dependency to keep thing simpler and just do `os.Exit(1)` instead.